istioctl/pkg/writer/envoy/configdump/configdump.go

```diff
@@ -133,3 +136,88 @@ func (c *ConfigWriter) PrintFullSummary(cf ClusterFilter, lf ListenerFilter, rf
 	}
 	return nil
 }
+
+// PrintVersionSummary prints version information for Istio and Envoy from the config dump
+func (c *ConfigWriter) PrintVersionSummary() error {
+	if c.configDump == nil {
+		return fmt.Errorf("config writer has not been primed")
+	}
+
+	bootstrapDump, err := c.configDump.GetBootstrapConfigDump()
+	if err != nil {
+		return err
+	}
+
+	var (
+		istioVersion, istioProxySha = c.getIstioVersionInfo(bootstrapDump)
+		envoyVersion                = c.getUserAgentVersionInfo(bootstrapDump)
+
+		tw = tabwriter.NewWriter(c.Stdout, 0, 8, 1, ' ', 0)
+	)
+
+	if len(istioVersion) > 0 {
+		fmt.Fprintf(tw, "Istio Version:\t%s\n", istioVersion)
+	}
+	if len(istioProxySha) > 0 {
+		fmt.Fprintf(tw, "Istio Proxy Version:\t%s\n", istioProxySha)
+	}
+	if len(envoyVersion) > 0 {
+		fmt.Fprintf(tw, "Envoy Version:\t%s\n", envoyVersion)
+	}
+
+	return tw.Flush()
+}
+
+func (c *ConfigWriter) getIstioVersionInfo(bootstrapDump *envoy_admin_v3.BootstrapConfigDump) (version, sha string) {
+	const (
+		istioVersionKey  = "ISTIO_VERSION"
+		istioProxyShaKey = "ISTIO_PROXY_SHA"
+	)
+
+	md := bootstrapDump.GetBootstrap().GetNode().GetMetadata().GetFields()
```

Comment:
This is neat. 
![image](https://user-images.githubusercontent.com/2920003/121552423-15eb9880-ca2e-11eb-8de8-47755a453713.png)